spec/stake/arithmetic.md

```diff
@@ -0,0 +1,59 @@
+# Fixed-Point Arithmetic for Rate Computation
+
+To compute base reward rate, base exchange rate, validator-specific exchange
+rates, and total validator voting power, we need to carefully perform
+arithmetic to avoid issues with precision and rounding. We use explicitly
+specified fixed-precision arithmetic for this, with a precision of 8 digits.
+This allows outputs to fit in a u64, with all products fitting in the output
+and plenty of headroom for additions.
+
+All integer values should be interpreted as unsigned 64-bit integers, with the
+exception of the validator's commission rate, which is a `u16` specified in
+terms of basis points.
+
+## Base Reward Rate
+
+The base reward is an input to the protocol, and the exact details of how this
+base rate $r_{e}$ is determined is not yet decided. For now, we can assume it is
+derived from the block header. $r_{e}$ is a fixed-precision `u64` integer with
+8 digits of precision.
+
+
+## Base Exchange Rate
+
+The base exchange rate, $\psi(e)$, can be safely computed as follows:
+
+$$\psi(e) = \prod_{0 \leq i < e} \lfloor \frac{(1e8 + r_i)}{1e8} \rfloor $$
```

Comment:
Should this be something like
```
$$\psi(e) = \left\lfloor \frac {\psi(e-1) (10^8 + r_e)} {10^8} \right\rfloor$$
```
instead? The existing formula doesn't seem quite right, because (1) if we apply the rounding to the rate term, we'll be rounding to 1 and then taking a product of 1s and (2) we know we have to compute the rate iteratively in order to have consistent rounding behavior.

Reply:
good catch, I actually ended up implementing it this way in python but the change didn't make it back into the notation here